src/Security/Authentication.php

```diff
@@ -23,18 +23,28 @@ class Authentication
     /** @var string */
     protected $password;
 
+    /** @var string */
+    protected $accessToken;
+
+    /** @var string */
+    protected $refreshToken;
+
     /**
-     * @param string $clientId
-     * @param string $secret
-     * @param string $username
-     * @param string $password
+     * @param string      $clientId
+     * @param string      $secret
+     * @param string      $username
+     * @param string      $password
+     * @param string|null $accessToken
+     * @param string|null $refreshToken
      */
-    public function __construct($clientId, $secret, $username, $password)
+    public function __construct($clientId, $secret, $username, $password, $accessToken = null, $refreshToken = null)
```

Comment:
username and password can be null. Two named constructors seems better (http://verraes.net/2014/06/named-constructors-in-php/): one for username/password `fromPassword`, one for access token/refresh token `fromToken`.

And spec for these constructors.